src/bmsg.c

```diff
@@ -585,6 +601,73 @@ int main (int argc, char *argv [])
             die ("unknown type %s", argv[argn]);
         }
     }
+    else
+    if (streq (bmsg_command, "request")) {
+
+        char *agent_name = argv[argn];
+        if (!agent_name)
+            die ("missing agent name", NULL);
+
+        argn ++;
+        char *subject = argv[argn];
+        if (!subject)
+            die ("missing subject", NULL);
+
+        mlm_client_set_producer (client, subject);
+
+        zmsg_t *msg = zmsg_new ();
+        zmsg_addstr (msg, "GET");
+
+        argn ++;
+        // Process additional arguments
+        while (argv [argn]) {
+            char *add_arg = argv [argn];
+            zmsg_addstr (msg, add_arg);
+            if ( verbose )
+                zsys_debug ("param: '%s'", add_arg);
+            argn ++;
+        }
+        if ( verbose ) {
+            zsys_debug ("Subject: %s", subject);
+            zmsg_print (msg);
+        }
+        int rv = mlm_client_sendto (client, agent_name, subject, NULL, 5000, &msg);
+        if (rv != 0)
+            die ("Request failed", NULL);
+
+        zmsg_destroy (&msg);
+
+        // Listen for the result
+        zpoller_t *poller = zpoller_new (mlm_client_msgpipe (client), NULL);
+        zsock_t *which = (zsock_t *) zpoller_wait (poller, timeout);
+        if (which == mlm_client_msgpipe (client)) {
+            msg = mlm_client_recv (client);
+
+            const char *sender = mlm_client_sender (client);
+            const char *subj = mlm_client_subject (client);
+            if (streq (sender, agent_name)) {
+                if (streq (subj, subject)) {
+                    if ( verbose ) {
+                        zsys_debug ("Got reply");
+                        zmsg_print (msg);
+                    }
+                    char *okfail = zmsg_popstr (msg);
+                    if ( okfail && streq (okfail, "OK")) {
+                        char *rc = zmsg_popstr(msg);
+                        while (rc) {
```

Comment:
As you already print the replied message (zmsg_print l652), I don't see the interest to print it again